testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -5030,6 +5030,14 @@ public void testMergeAllColumnsReversed()
         assertUpdate("DROP TABLE " + targetTable);
     }
 
+    @Test
+    public void testDropTableIfExistsStatementWithNonExistingObject()
+    {
+        assertUpdate("DROP TABLE IF EXISTS non_existing_catalog.non_existing_schema.non_existing_table");
+        assertUpdate("DROP TABLE IF EXISTS non_existing_schema.non_existing_table");
+        assertUpdate("DROP TABLE IF EXISTS non_existing_table");
```

Comment:
`testDropTableIfExists` already covers this one.

Reply:
Is that okay to merge all statements in `testDropTableIfExists` or we can have separate tests for non-existing catalog and schema?

Reply:
> for non-existing catalog and schema

No need to add non-existing catalog case to BaseConnectorTest in my opinion. You can just add non-existing schema case to `testDropTableIfExists`. A separated test method is also acceptable, but the place should be closed to `testDropTableIfExists`.

Reply:
Moved test cases to `TestDropTableTask`. Please see https://github.com/trinodb/trino/pull/15464#discussion_r1054386010